3.next - Use CookieCollection in Client responses

markstory opened the pull request:
Use the new CookieCollection objects in Client responses. This allows us to remove one copy of the Cookie parsing code we had.

Refs #10406

markstory committed:
Add Cookie::toArray()

This method belongs here as all the data is here.

markstory committed:
Add toArrayCompat()

This method is necessary as the existing Http\Client code expects the
original date format to be returned. Having this method allows us to
have less duplication across the framework.

markstory committed:
Add a factory method for building a CookieCollection from a header.

markstory committed:
Use CookieCollection internally in Client responses.

markstory committed:
Expose the CookieCollection in a response.

Allow developer to interact with a response's CookieCollection if they
want to.

lorenzo commented on src/Http/Client/Response.php:
```diff
@@ -453,11 +434,43 @@ public function getCookie($name)
+     *
+     * @return array Array of Cookie data.
+     */
+    protected function _getCookies()
```
Why have a this method if the public method is supposed to do the same as the public method with the same name? Maybe just make them the same method?

markstory replied:
This is for the magic properties that the response exposes. `$response->cookies` is provided by this method and the _get` prefix is part of the required convention for exposing properties through methods on this class.

ravage84 replied:
@markstory May be add ``This is for the magic properties that the response exposes. $response->cookies is provided by this method.`` or something like that as the long description of this method. This way it becomes clear.

markstory committed:
Fix incorrect test.

markstory committed:
Use an alias to avoid duplicate class name issues.

The deprecated CookieCollection shares a name with this imported class
so we need an alias.
